core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingCurves.kt

```diff
@@ -344,21 +346,38 @@ private fun computeIndicationBrakingCurveFromRef(
 
 /**
  * Keep the part of the full braking curve which is located underneath the overlay and intersects
- * with it or with begin position. If the part has no intersection, return null. Ex: the part only
- * has negative positions, and the overlay starts at 0.0.
+ * with it or with begin position. If the part has no intersection, return null.
  */
 private fun keepBrakingCurveUnderOverlay(
-    fullBrakingCurve: EnvelopePart,
+    fullBrakingCurve: Envelope,
     overlay: Envelope,
     beginPos: Double
 ): EnvelopePart? {
     if (fullBrakingCurve.endPos <= beginPos) {
+        etcsBrakingCurvesLogger.warn(
+            "The ETCS braking curve ending at ($beginPos, ${fullBrakingCurve.endSpeed}) does not intersect with the overlay envelope."
```

Comment:
nit (as we had a misunderstanding about that in comments 😅):
```suggestion
            "The position-range of ETCS braking curve ending at ($beginPos, ${fullBrakingCurve.endSpeed}) does not intersect with the overlay envelope's position-range."
```